File: mobile/lib/models/category.dart
class AppCategory {
  final int id;
  final String name;
  final bool isActive;
  final int sortOrder;
  final String? createdAt;
  final String? updatedAt;

  const AppCategory({
    required this.id,
    required this.name,
    required this.isActive,
    required this.sortOrder,
    this.createdAt,
    this.updatedAt,
  });

  factory AppCategory.fromJson(Map<String, dynamic> json) {
    return AppCategory(
      id: json['id'],
      name: json['name'],
      isActive: json['is_active'] ?? true,
      sortOrder: json['sort_order'] ?? 0,
      createdAt: json['created_at'],
      updatedAt: json['updated_at'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'name': name,
      'sort_order': sortOrder,
    };
  }
}

File: backend/app/models/category.py
from sqlalchemy import Column, Integer, String, Boolean, DateTime
from datetime import datetime
from app.database import Base


class Category(Base):
    """Dynamic expense categories with soft-delete and ordering."""
    __tablename__ = "categories"

    id = Column(Integer, primary_key=True, index=True)
    name = Column(String, unique=True, nullable=False)
    color = Column(String, default="#9E9E9E", nullable=False)
    icon = Column(String, nullable=True)
    type = Column(String, default="expense", nullable=False)
    is_active = Column(Boolean, default=True, nullable=False)
    sort_order = Column(Integer, default=0, nullable=False)
    created_at = Column(DateTime, default=datetime.utcnow)
    updated_at = Column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow)
